src-input/duk_debugger.c

```diff
@@ -1494,33 +1480,48 @@ DUK_LOCAL void duk__debug_handle_eval(duk_hthread *thr, duk_heap *heap) {
 	 * activation.  For now, use global object eval() function, with the eval
 	 * considered a 'direct call to eval'.
 	 *
-	 * Callstack level for debug commands only affects scope -- the callstack
+	 * callstack index for debug commands only affects scope -- the callstack
 	 * as seen by, e.g. Duktape.act() will be the same regardless.
 	 */
 
-	/* nargs == 2 so we can pass a callstack level to eval(). */
+	/* nargs == 2 so we can pass a callstack index to eval(). */
 	duk_push_c_function(ctx, duk_bi_global_object_eval, 2 /*nargs*/);
 	duk_push_undefined(ctx);  /* 'this' binding shouldn't matter here */
 
-	(void) duk_debug_read_hstring(thr);
-	if (duk_debug_peek_byte(thr) != DUK_DBG_IB_EOM) {
```

Comment:
Ah, right: `duk_tval` is a borrowed reference and must be left on the  value stack to prevent premature collection.

But the lookahead solution is a bit cleaner, so I'll just do that.